File: tests/snapshots/iterator_more.rs
// More iterator links — take, skip, rev — and the Option-returning consumers
// (max, position) that pair with Match.

fn main() {
    let hi: i64 = 8;
    let mut nums: Vec<i64> = Vec::new();
    for i in 1..=hi {
        nums.push(i);
    }
    let firstthree: Vec<i64> = nums.iter().copied().take(3).collect();
    let lasttwo: Vec<i64> = nums.iter().copied().rev().take(2).collect();
    let tail: Vec<i64> = nums.iter().copied().skip(5).collect();
    for n in &firstthree {
        println!("take: {}", *n);
    }
    for n in &lasttwo {
        println!("rev:  {}", *n);
    }
    for n in &tail {
        println!("skip: {}", *n);
    }
    match nums.iter().copied().max() {
        Some ( top ) => {
            println!("max:  {}", top);
        }
        None => {
            println!("empty");
        }
    }
    match nums.iter().copied().position(|x| x > 6) {
        Some ( idx ) => {
            println!("pos:  {}", idx);
        }
        None => {
            println!("none over 6");
        }
    }
}

File: tests/snapshots/iterator_more.py
# More iterator links — take, skip, rev — and the Option-returning consumers
# (max, position) that pair with Match.

from dataclasses import dataclass

@dataclass
class Some:
    value: object

def _vb(x):
    if isinstance(x, bool):
        return "true" if x else "false"
    if isinstance(x, float) and x.is_integer():
        return str(int(x))
    return str(x)

def main():
    hi: int = 8
    nums: list[int] = []
    for i in range(1, hi + 1):
        nums.append(i)
    firstthree: list[int] = nums[:3]
    lasttwo: list[int] = nums[::-1][:2]
    tail: list[int] = nums[5:]
    for n in firstthree:
        print(f"take: {_vb(n)}")
    for n in lasttwo:
        print(f"rev:  {_vb(n)}")
    for n in tail:
        print(f"skip: {_vb(n)}")
    _m0 = (Some(max(nums)) if nums else None)
    match _m0:
        case Some(top):
            print(f"max:  {_vb(top)}")
        case None:
            print('empty')
    _m1 = next((Some(_i) for _i, x in enumerate(nums) if x > 6), None)
    match _m1:
        case Some(idx):
            print(f"pos:  {_vb(idx)}")
        case None:
            print('none over 6')


if __name__ == "__main__":
    main()
